### PR TITLE
migrate to aioapns #10

### PR DESCRIPTION
**Pull request by patroqueeet**
what do you think? would that work? (disclaimer: not having that much async experience)

### CODE REVIEW

**Comment by alexandernst** [comment]
Can you revert the styling changes so that it can be easier to review the PR? 🙏

**Comment by patroqueeet** [comment]
@alexandernst hey, sry was aware of it. but the hour was too late to have some energy for the reversal. but I'd prefer to convince you of applying `black` project wide. It's really some kind of standard now, because it standardises format and devs don't have to worry anymore.. I can send you a black'd project PR first, and then we do #15.

I'd be really happy about a black'd code base :) (and maybe even imports standardised with `isort` ) - what do you think?

**Comment by patroqueeet** [comment]
@alexandernst seen my comment?

**Comment by alexandernst** [comment]
@patroqueeet Sorry for the delayed response, too much work, I just couldn't reply earlier.

I'll have to reject the whole `black` refactoring. I'm sure tons of people like it, but I can't stand some of it's indenting decisions, and since `black` is `“Any color you like, as long as it's black”` I'd rather not have it.

The imports sorted with `isort` sounds good, but I'd also rather have it in another PR. Split the changes so we can review the actual code changes.

**Comment by patroqueeet** [comment]
@alexandernst un-`black`'d & un-`isort`ed ✅

File: django_walletpass/services.py [hunk 1]
```diff
@@ -1,70 +1,46 @@
...
-            self.get_credentials(),
-            push_sandbox=WALLETPASS_CONF['PUSH_SANDBOX'],
-            pass_type_id=WALLETPASS_CONF['PASS_TYPE_ID'],
-            push_token=token,
+        client = APNs(
+            key=WALLETPASS_CONF["TOKEN_AUTH_KEY_PATH"],
+            key_id=WALLETPASS_CONF["TOKEN_AUTH_KEY_ID"],
+            team_id=WALLETPASS_CONF["TEAM_ID"],
+            topic=WALLETPASS_CONF["PASS_TYPE_ID"],
+            use_sandbox=WALLETPASS_CONF["PUSH_SANDBOX"],
```
Comment (alexandernst) [inline django_walletpass/services.py:38]: If I read this correctly, it won't be possible to use cert/key auth (legacy) anymore, right?

Comment (patroqueeet) [reply django_walletpass/services.py:38]: this is correct. I had to switch to token based anyway, because the old way stopped working.

**Comment by patroqueeet** [comment]
@alexandernst next steps? pls advice!

**Comment by alexandernst** [comment]
I think this is good to go! We should release a major version since we're dropping the token based auth.

@NachE are you good with these changes?

**Comment by patroqueeet** [comment]
@alexandernst as your good with these changes, let me do some more QA (I had in mind) before it's being released, ok?

**Comment by alexandernst** [comment]
Sounds good!

**Comment by NachE** [comment]
> I think this is good to go! We should release a major version since we're dropping the token based auth.
> 
> @NachE are you good with these changes?

Agree 👍

**Comment by patroqueeet** [comment]
@alexandernst @NachE did some real live staging env testing. I all looked good, so we have some good proof that the change is fine. we should may clean up the settings, when deprecating the cert based auth approach later on. made a ticket for this.

should we merge it?

**Comment by alexandernst** [comment]
@patroqueeet Just one more thing. Update the changelog, stating that this is a breaking change for users using token auth. And increment the version in the `setup.py`

**Comment by patroqueeet** [comment]
> @patroqueeet Just one more thing. Update the changelog, stating that this is a breaking change for users using token auth. And increment the version in the `setup.py`

@alexandernst 
done. and updated py version to 3.6. as this is the oldest version `aioapns` is tested with

File: setup.py [hunk 2]
```diff
@@ -4,49 +4,49 @@
...
     README = ffile.read()
 
 setup(
-    name='django-walletpass',
-    python_requires='>=3.5.0',
-    version='2.0',
-    author='Develatio Technologies S.L.',
-    author_email='contacto@develat.io',
+    name="django-walletpass",
+    python_requires=">=3.6.0",
```
Comment (alexandernst) [inline setup.py]: You should add this to the CHANGELOG as well. If we're increasing the min py version we should specify it.

Comment (patroqueeet) [reply setup.py]: done

**Pull request merged**